src/main/java/com/nashtech/service/impl/ReactiveDataServiceImpl.java

```diff
@@ -6,27 +6,72 @@
 import com.nashtech.service.ReactiveDataService;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClientException;
 import reactor.core.publisher.Flux;
 
 /**
- * Service implementation class for performing reactive data access
- * operations on cars.
- * This service interacts with the {@link CloudDataService}
- * to retrieve car data in a reactive manner.
+ * Service class for handling vehicle-related operations.
  */
-@Service
 @Slf4j
-public class ReactiveDataServiceImpl implements ReactiveDataService {
+@Service
+public class ReactiveDataServiceImpl implements
+        ReactiveDataService {
 
     /**
-     * The reactive Service for {@link Car} entities
-     * in Cosmos DB.
-     * Used for performing CRUD operations and reactive data access.
+     * WebClient instance for making HTTP requests to the external API.
+     */
+    @Autowired
+    private WebClient webClient;
+
+    /**
+     * URL of the external API for retrieving vehicle data.
+     */
+    @Value("${apiUrl}")
+    private String apiUrl;
+
+    /**
+     * Autowired instance of CloudDataService
+     * used for publishing vehicle data to Google Cloud Pub/Sub.
      */
     @Autowired
     private CloudDataService cloudDataService;
 
+
+    /**
+     * Retrieves vehicle data from an external API.
+     * @throws WebClientException If an error occurs during
+     * data retrieval from the external API.
+     */
+    public void fetchAndSendData() {
+        webClient.get()
+                .uri(apiUrl)
+                .retrieve()
+                .bodyToFlux(Car.class)
+                .switchIfEmpty(Flux.defer(() -> {
+                 log.info("No data found");
+                 return Flux.empty();
+                }))
+                .onErrorResume(throwable -> {
+                    log.error("Error occurred during data retrieval: "
+                    + throwable.getMessage());
+                    return Flux.error (
+                            new WebClientException (
+                                    "Failed to retrieve car data", throwable) { });
+                })
+               .subscribe(
+                        data -> {
+                            try {
```

Comment:
Remove try catch from lambda once exceptions are handled in CloudDataService implementations